From 073e90671d34b18b631be031936a5adb87a85c9a Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Fri, 2 Aug 2013 15:34:47 -0700
Subject: [PATCH] Remove last dm partition (#851705)

Allow _dm_reread_part_table to remove the last partition. The check for
largest_partnum stopped it from removing the dm partition device if it
was the only one.

This check was added in commit 1742b051d78493b90bcb801e68a2be0277bcf72f
to protect code that no longer exists.
---
 libparted/arch/linux.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/libparted/arch/linux.c b/libparted/arch/linux.c
index d735712..5a17d3b 100644
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -2522,9 +2522,6 @@ static int
 _dm_reread_part_table (PedDisk* disk)
 {
         int largest_partnum = ped_disk_get_last_partition_num (disk);
-        if (largest_partnum <= 0)
-          return 1;
-
         int     rc = 1;
         int     i;
 
-- 
1.8.3.1

